public/app/features/plugins/admin/hooks/usePluginDetailsTabs.tsx

```diff
@@ -1,104 +1,144 @@
+import { GrafanaPlugin, NavModelItem, PluginIncludeType, PluginType } from '@grafana/data';
 import { config } from '@grafana/runtime';
+import { toIconName } from '@grafana/ui';
 import { contextSrv } from 'app/core/core';
 import { AccessControlAction } from 'app/types';
 
+import { PluginDetailsHeader } from '../components/PluginDetailsHeader';
 import { usePluginConfig } from '../hooks/usePluginConfig';
 import { isOrgAdmin } from '../permissions';
-import { CatalogPlugin, PluginDetailsTab, PluginTabIds, PluginTabLabels } from '../types';
+import { CatalogPlugin, PluginTabIds, PluginTabLabels } from '../types';
 
 type ReturnType = {
   error: Error | undefined;
   loading: boolean;
-  tabs: PluginDetailsTab[];
-  defaultTab: string;
+  navModel: NavModelItem;
+  activePageId: PluginTabIds | string;
 };
 
-export const usePluginDetailsTabs = (plugin?: CatalogPlugin, defaultTabs: PluginDetailsTab[] = []): ReturnType => {
+export const usePluginDetailsTabs = (plugin?: CatalogPlugin, pageId?: PluginTabIds): ReturnType => {
   const { loading, error, value: pluginConfig } = usePluginConfig(plugin);
-  const isPublished = Boolean(plugin?.isPublished);
   const { pathname } = useLocation();
+  const defaultTab = useDefaultPage(plugin, pluginConfig);
+  const parentUrl = pathname.substring(0, pathname.lastIndexOf('/'));
+  const isPublished = Boolean(plugin?.isPublished);
 
-  const [tabs, defaultTab] = useMemo(() => {
+  const currentPageId = pageId || defaultTab;
+  const navModelChildren = useMemo(() => {
     const canConfigurePlugins =
       plugin && contextSrv.hasAccessInMetadata(AccessControlAction.PluginsWrite, plugin, isOrgAdmin());
-    const tabs: PluginDetailsTab[] = [...defaultTabs];
-    let defaultTab;
+    const navModelChildren: NavModelItem[] = [];
     if (isPublished) {
-      tabs.push({
-        label: PluginTabLabels.VERSIONS,
-        icon: 'history',
+      navModelChildren.push({
+        text: PluginTabLabels.VERSIONS,
         id: PluginTabIds.VERSIONS,
-        href: `${pathname}?page=${PluginTabIds.VERSIONS}`,
+        icon: toIconName('history'),
+        url: `${pathname}?page=${PluginTabIds.VERSIONS}`,
+        active: PluginTabIds.VERSIONS === currentPageId,
       });
     }
 
     // Not extending the tabs with the config pages if the plugin is not installed
     if (!pluginConfig) {
-      defaultTab = PluginTabIds.OVERVIEW;
-      return [tabs, defaultTab];
+      return navModelChildren;
     }
 
     if (config.featureToggles.panelTitleSearch && pluginConfig.meta.type === PluginType.panel) {
-      tabs.push({
-        label: PluginTabLabels.USAGE,
+      navModelChildren.push({
+        text: PluginTabLabels.USAGE,
         icon: 'list-ul',
         id: PluginTabIds.USAGE,
-        href: `${pathname}?page=${PluginTabIds.USAGE}`,
+        url: `${pathname}?page=${PluginTabIds.USAGE}`,
+        active: PluginTabIds.USAGE === currentPageId,
       });
     }
 
     if (canConfigurePlugins) {
```

Comment:
nit: what about doing an early exit here? basically change this to something like:

```ts
if (!canConfigurePlugins) {
   return navModelChildren;
}
```

To prevent nesting.

Reply:
Nice suggestion, have updated it. 👍